ports/curl/portfile.cmake

```
@@ -109,5 +113,14 @@ else()
     )
 endif()
 
+# Fix the pkgconfig file for debug
+if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
+    if(VCPKG_TARGET_IS_WINDOWS)
+        vcpkg_replace_string(${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/libcurl.pc "-lcurl" "-lcurl-d")
+        vcpkg_replace_string(${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/libcurl.pc "/lib/zlib.lib" "/debug/lib/zlibd.lib")	
+    endif()
+    file(COPY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/libcurl.pc DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig)
+endif()
+
```

add the call after you corrected the *.pc files
```suggestion
vcpkg_fixup_pkgconfig()
```

---

Hello, I took into account both your request and the one from @PhoebeHui to change :/vcpkg/clean/vcpkg/installed/x64-windows/lib/zlib.lib style lib to -lzlib. Actually, I had to do it otherwise vcpkg_fixup_pkgconfig() call was failing.
I also added other libs for some feature enabled builds such as :
vcpkg.exe install curl[core,tool,http2,ssh,openssl,winssl,mbedtls,sspi]:x64-windows